site/cds_rdm/migration/extract.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022 CERN.
+#
+# CDS-RDM is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+
+"""CDS-RDM migration extract module."""
+
+from os import listdir
+from os.path import isfile, join
+from pathlib import Path
+
+import click
+import json
+
+from invenio_rdm_migrator.extract import Extract
+
+
+class LegacyExtract(Extract):
```

Comment:
not sure if this reusable. 
In our case it iterates through many files containing dump chunks (1000 records per file).
In case of Zenodo, if I understand correctly, they plan to use a dump query. 
I think it makes sense to use separate custom extract classes, individually per instance. At least at this point, I would not try to make one solution for all